api/src/main/java/org/openmrs/module/attachments/obs/AbstractDocumentHandler.java

```diff
@@ -61,6 +61,10 @@ protected void setParent(ComplexObsHandler complexObsHandler) {
   final protected ComplexObsHandler getParent() {
     return parent;
   }
+  
+  protected static String buildSmallFileFileName(String fileName) {
+      return FilenameUtils.removeExtension(fileName) + "_smallfile" + "." + FilenameUtils.getExtension(fileName);
+  }
 
   protected static String buildThumbnailFileName(String fileName) {
```

Comment:
**ATTENTION**: It is important that the check searches for a _suffix_ and not for the file name to just contain the special string. This is because we cannot ever be sure that a file containing the special string might ever be saved in the first place (in fact it is totally possible). So basically: `StringUtils.endsWith(str, suffix)` ;-)